### PR TITLE
Add missing tests for TaskService and minor fix for DTO conversion

### PR DESCRIPTION
**Pull request by Beacze**
Fix NPE on processing user fetching.

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/services/data/TaskService.java [hunk 1]
```diff
@@ -358,7 +358,9 @@ private TaskDTO convertRelatedJSONObjects(JSONObject jsonObject, TaskDTO taskDTO
         Integer process = getIntegerPropertyForDTO(jsonObject, "process");
         taskDTO.setProcess(serviceManager.getProcessService().findById(process, true));
         Integer processingUser = getIntegerPropertyForDTO(jsonObject, "processingUser");
-        taskDTO.setProcessingUser(serviceManager.getUserService().findById(processingUser, true));
+        if (processingUser != 0) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: If no user is currently working on this task as he / she aborted this task, this should be reflected into the data.

Comment (Beacze) [reply Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: That's exactly what this line does. It checks if some user works on this, if yes, it requests information about this user, if not, it leaves this attribute null.

Comment (henning-gerhardt) [reply Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: Sorry, I did not see that you are resetting the user. Your code set only the code if the `processingUser` is not `0` and if I look into `getIntegerPropertyForDTO`  it is returning `0` if the value even contains value `null`. So it is never set to an other value even no user is working any more on this task.

Comment (Beacze) [reply Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: Could you explain more detailed what is a problem here?

Comment (henning-gerhardt) [reply Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: I would only ensure that a formally attached user to a task is really removed from indexed data if this user is not involved with this task any more.

Comment (Beacze) [reply Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: Method responsible for it is manageProcessingUserDependenciesForIndex(Task task), but I see that there must be actual check if ProcessingUser is not empty.

File: Kitodo/src/main/java/org/kitodo/services/data/TaskService.java [hunk 2]
```diff
@@ -117,7 +120,18 @@ private void manageProcessingUserDependenciesForIndex(Task task) throws CustomRe
...
             User user = task.getProcessingUser();
-            serviceManager.getUserService().saveToIndex(user);
+            if (user != null) {
+                serviceManager.getUserService().saveToIndex(user);
+            }
+            reSaveUserAfterRemoveFromProcessing(task);
+        }
+    }
+
+    private void reSaveUserAfterRemoveFromProcessing(Task task)
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: It is not more a re-index after removing as your method does apply changes only to index?

**Pull request merged**